serialization/cbor.ksy

```diff
@@ -0,0 +1,328 @@
+    seq:
+      - id: value
+        type:
+          switch-on: type
+          cases:
+            "type::simple": simple_or_float
+            #"type::f2": f2
+            "type::f4": f4
+            "type::f8": f8
+    instances:
+      type:
+        value: _parent.as<cbor>.ib.additional_info
+        enum: type
+    enums:
+      type:
+        20: "false"
+        21: "true"
+        22: "null"
+        23: "undefined"
+        24: simple
+        25: f2
+        26: f4
+        27: f8
+        31: chain_break
+  tag:
+    doc: "Marks the underlying CBOR structure to be transformed by decoder into the corresponding object. Should not cause security issues by itself since non-private tags are standardized and there is no dangerous tags there."
+    seq:
+      - id: tag_value
+        type: uint
+      - id: tagged
+        type: cbor
+    instances:
+      ib:
+        value: _parent.ib
+      tag:
+        value: tag_value.value
+        enum: tag
+      value:
+        pos: 0
+        type:
+          switch-on: tag
+          cases:
+            "tag::big_float": big_float
+            "tag::decimal": decimal
+    types:
+      big_float_common:
+        params:
+          - id: arr
+            type: array
+        instances:
+          exponent_valid:
+            value: "arr.length.value >= 1 and (arr.items[0].ib.major_type == type::uint or arr.items[0].ib.major_type == type::neg_uint)"
+          mantissa_valid:
+            value: "arr.length.value >= 2 and (arr.items[1].ib.major_type == type::uint or arr.items[1].ib.major_type == type::neg_uint)"
+            # bigint mantissa is not yet supported
+          exponent:
+            value: "arr.items[0].payload.as<uint>.value"
+            if: exponent_valid
+          mantissa:
+            value: "arr.items[1].payload.as<uint>.value"
+            if: mantissa_valid
+          valid:
+            value: exponent_valid and mantissa_valid
+      # Kaitai has neither inheritance no duck typing, so have to repeat code
+      big_float:
+        instances:
+          array_valid:
+            value: _parent.as<tag>.tagged.ib.major_type == type::array
+          arr:
+            value: _parent.as<tag>.tagged.payload.as<array>
+            if: array_valid
+          array_length_valid:
+            value: "array_valid and arr.length.value == 2"
+          bfc:
+            pos: 0
+            type: big_float_common(arr)
+            if: array_valid and array_length_valid
+          scale:
+            pos: 0
+            type: pow2(bfc.exponent)
+            # a dirty and limited workaround for Kaitai not having a power operation. The limitation is: exponent is only allowed to be 64 bits
+          float_approximation:
+            value: "bfc.mantissa*scale.value"
+            if: bfc.valid
+        types:
+          pow2:
+            params:
+              - id: log
+                type: u8
+            instances:
+              is_positive:
+                value: log>=0
```

Comment:
`is_positive` will always be `true`, because `log` is of type `u8`, which means unsigned integer.

Reply:
Nice catch, thanks.